hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/AvroConvertor.java

```diff
@@ -175,9 +176,12 @@ public GenericRecord withKafkaFieldsAppended(ConsumerRecord consumerRecord) {
     for (Schema.Field field :  record.getSchema().getFields()) {
       recordBuilder.set(field, record.get(field.name()));
     }
+    
+    String kafkaKey = String.valueOf(consumerRecord.key());
     recordBuilder.set(KAFKA_SOURCE_OFFSET_COLUMN, consumerRecord.offset());
     recordBuilder.set(KAFKA_SOURCE_PARTITION_COLUMN, consumerRecord.partition());
     recordBuilder.set(KAFKA_SOURCE_TIMESTAMP_COLUMN, consumerRecord.timestamp());
+    recordBuilder.set(KAFKA_SOURCE_KEY_COLUMN, kafkaKey);
```

Comment:
```suggestion
    recordBuilder.set(KAFKA_SOURCE_KEY_COLUMN, String.valueOf(consumerRecord.key()));
```

Reply:
accepted the suggestion & have made the changes